android/src/main/java/com/rogchap/react/modules/twilio/TwilioModule.java

```diff
@@ -146,7 +146,19 @@ public void onError(Exception e) {
                 }
             });
         } else {
-            sendEvent("deviceReady", null);
+            try {
+                if (_phone != null) {
+                    _phone.release();
+                }
+                _phone = Twilio.createDevice(token, dl);
```

Comment:
go ahead and refactor ! I don't really "speak" java so tried to minimize impact ;)
I added this since the functionality of calling 'initWithToken' multiple times is essentially broken here.
The first time (maybe) works (sometimes not for me, not sure why). 
The second time simply does nothing since Twilio.isInitialized() is true. 

I'm not sure why (but maybe could be a clue for you) - I noticed that often _phone is null even after initialization, which is weird... In any case - since I always 'initWithToken' before I use the device (to get fresh permissions), the patch here fixes the issue for me.

btw - thanks for helping out with the android part! it's much appreciated !